src/runtime/setup-session.js

```diff
@@ -1,9 +1,16 @@
 import Session from './session';
+import { warn } from '../shared/log';
 
 export default function setupSession(global) {
-
   // The name of the property encodes the session/node compatibilty version
   if (!global._heimdall_session_3) {
     global._heimdall_session_3 = new Session();
   }
+
+  for (let i = 1; i < 3; i++) {
```

Comment:
Might be overly unnecessary, but we could do `const CURRENT_VERSION = 3;` in the module scope and replace `3` everywhere with the constant so that future upgrades are easy.